File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/cn/porkchop/pinyougou/sellergoods/service/SpecificationService.java
package cn.porkchop.pinyougou.sellergoods.service;

import cn.porkchop.pinyougou.pojo.PageResult;
import cn.porkchop.pinyougou.pojo.TbSpecification;
import cn.porkchop.pinyougou.pojo.pojogroup.Specification;

import java.util.List;
import java.util.Map;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface SpecificationService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbSpecification> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    PageResult findByPagination(int pageNum, int pageSize);


    /**
     * 增加
     */
    void add(Specification specification);


    /**
     * 修改
     */
    void update(Specification specification);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    Specification findById(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum
     *         当前页 码
     * @param pageSize
     *         每页记录数
     * @return
     */
    PageResult findWithConditionsByPagination(TbSpecification specification, int pageNum, int pageSize);

    /**
     * 查询所有
     *
     * @date 2018/6/17 23:31
     * @author porkchop
     */
    List<Map<String, String>> findAllForSelect2();

}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/specificationService.js
//服务层
app.service('specificationService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../specification/findAll.php');
    };
    //分页
    this.findByPagination = function (page, rows) {
        return $http.get('../specification/findByPagination.php?page=' + page + '&rows=' + rows);
    };
    //查询实体
    this.findById = function (id) {
        return $http.get('../specification/findById.php?id=' + id);
    };
    //增加
    this.add = function (entity) {
        return $http.post('../specification/add.php', entity);
    };
    //修改
    this.update = function (entity) {
        return $http.post('../specification/update.php', entity);
    };
    //删除
    this.dele = function (ids) {
        return $http.get('../specification/delete.php?ids=' + ids);
    };
    //搜索
    this.findWithConditionsByPagination = function (page, rows, searchEntity) {
        return $http.post('../specification/findWithConditionsByPagination.php?page=' + page + "&rows=" + rows, searchEntity);
    };
    this.findAllForSelect2=function () {
        return $http.get('../specification/findAllForSelect2.php');
    }
});
